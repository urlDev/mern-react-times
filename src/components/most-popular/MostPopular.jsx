import React from "react";
import { useSelector } from "../../utils/react-redux-hooks";

import NewsPlaceholder from "../../assets/newsPlaceholder.png";

import {
  MostPopularContainer,
  PopularTitle,
  PopularStoriesContainer,
  StoryContainer,
  StoryTitle,
  ImageContainer,
  SeeAll,
  MostPopularWithHeader,
} from "./MostPopular.styles";

const MostPopular = () => {
  const { popular, width } = useSelector((news) => news.news);

  // Wanted to show random results on each refresh/load
  // but random makes the app loads more than once
  // const random = Math.floor(Math.random() * popular.length);

  return (
    <MostPopularWithHeader>
      <PopularTitle>Most Popular</PopularTitle>
      <MostPopularContainer>
        {popular.length > 0
          ? popular.map((story) => (
              <PopularStoriesContainer
                key={story.id}
                href={story.url}
                target="_blank"
              >
                <StoryContainer>
                  <StoryTitle font="var(--font-header)">
                    {width > 830
                      ? story.title.split(" ").slice(0, 6).join(" ")
                      : story.title}
                    ...
                  </StoryTitle>
                  <StoryTitle color="var(--gray)">
                    {width > 830
                      ? story.abstract.split(" ").slice(0, 10).join(" ")
                      : story.abstract}
                    ...
                  </StoryTitle>
                </StoryContainer>
                <ImageContainer
                  background={
                    story.media[0]
                      ? story.media[0]["media-metadata"][2].url
                      : NewsPlaceholder
                  }
                />
              </PopularStoriesContainer>
            ))
          : null}
        <SeeAll
          popular="43px"
          href="https://www.nytimes.com/trending/?action=click&contentCollection=Africa&module=MostEmailed&pgtype=article&region=Marginalia&src=me&version=Full"
          target="_blank"
        >
          See All
        </SeeAll>
      </MostPopularContainer>
    </MostPopularWithHeader>
  );
};

export default MostPopular;
