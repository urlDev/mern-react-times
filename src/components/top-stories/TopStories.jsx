import React from "react";
import moment from "moment";
import { useSelector } from "utils/react-redux-hooks";

import StoryTopic from "components/story-topic/StoryTopic";
import placeholder from "assets/newsPlaceholder.png";

import { Date } from "components/story-topic-headers/StoryTopicHeaders.styles";
import { SeeAll } from "components/most-popular/MostPopular.styles";
import {
  TopStoriesContainer,
  StoryContainer,
  Tag,
  Title,
  SubTitle,
  TagContainer,
} from "./TopStories.styles";

const TopStories = () => {
  const { story } = useSelector((state) => state.news);

  return (
    <div>
      <StoryTopic />
      <TopStoriesContainer>
        {story.slice(0, 3).map((data) => {
          const url = data?.multimedia?.[0]?.url;

          return (
            <StoryContainer key={data.title}>
              <img
                src={url ?? placeholder}
                alt="story"
                style={{ width: !url ? "400px" : "100%" }}
              />
              <TagContainer>
                <h1>
                  <Tag>{data.section}</Tag>
                </h1>
                <Date margin="10px" width="30%">
                  {moment(data.created_date).format("MMM DD, YYYY")}
                </Date>
              </TagContainer>

              <Title header={data.title.split(" ")}>{data.title}</Title>
              <SubTitle>
                {data.abstract.split(" ").slice(0, 20).join(" ")}...
              </SubTitle>
              <SeeAll href={data.url} target="_blank" rel="noopener">
                Continue Reading
              </SeeAll>
            </StoryContainer>
          );
        })}
      </TopStoriesContainer>
    </div>
  );
};

export default TopStories;
