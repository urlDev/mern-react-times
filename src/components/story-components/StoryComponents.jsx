import React from 'react';
import { useSelector } from 'react-redux';

import ErrorFallback from '../error-fallback/ErrorFallback';
import TopStories from '../top-stories/TopStories';
import MostPopular from '../most-popular/MostPopular';

import { StoryComponentsContainer } from './StoryComponents.styles';

const Home = () => {
  const { errorNews } = useSelector((news) => news.news);

  console.log(errorNews);

  return (
    <>
      <StoryComponentsContainer>
        <TopStories />
        <MostPopular />
      </StoryComponentsContainer>
    </>
  );
};

export default Home;
