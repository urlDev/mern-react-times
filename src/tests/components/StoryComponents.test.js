import { shallow } from "enzyme";

import StoryComponents from "components/story-components/StoryComponents";
import TopStories from "components/top-stories/TopStories";
import MostPopular from "components/most-popular/MostPopular";

describe("Testing StoryComponents", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<StoryComponents />);
  });

  test("Should render the components", () => {
    expect(wrapper).toHaveLength(1);
    expect(wrapper.containsMatchingElement(<TopStories />)).toBeTruthy();
    expect(wrapper.containsMatchingElement(<MostPopular />)).toBeTruthy();
  });
  test("Should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
