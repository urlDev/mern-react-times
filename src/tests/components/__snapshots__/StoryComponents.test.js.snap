// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing StoryComponents Should match the snapshot 1`] = `
<ErrorBoundary
  FallbackComponent={[Function]}
>
  <styled.div>
    <TopStories />
    <MostPopular />
  </styled.div>
</ErrorBoundary>
`;
